aiokafka/consumer/group_coordinator.py

```diff
@@ -1202,46 +1217,67 @@ async def perform_group_join(self):
                 metadata = metadata.encode()
             group_protocol = (assignor.name, metadata)
             metadata_list.append(group_protocol)
+            # for KIP-394 we may have to send a second join request
```

Comment:
This sounds like a completely different effort, maybe worth separating out to another PR?

Reply:
The bulk of this addition is captured in the new `while` loop [wrapping lines 1222-1279](https://github.com/joshuaherrera/aiokafka/blob/kip-345/aiokafka/consumer/group_coordinator.py#L1220-L1279), and the newly added [MemberIdRequired error](https://github.com/joshuaherrera/aiokafka/blob/kip-345/aiokafka/errors.py#L859). [KIP-394](https://cwiki.apache.org/confluence/display/KAFKA/KIP-394%3A+Require+member.id+for+initial+join+group+request) introduces the `MemberIdRequired` error, and according to the KIP, if the client gets this error they should retry the join with the previous member id.

I'd prefer to leave it in since the [original author](https://github.com/robinhood/aiokafka/commit/cd44bdcf57b4aa6f675783ed1043b2d643b0ac54) thought it necessary and the KIP itself calls out that this is "an important complement to KIP-345" but if you think this should be moved to it's own PR (and would delay this PR) I can remove the code related to KIP-394 from this PR.

Reply:
@tvoinarovskyi Any updates on next steps? If you or @ods think it's best to remove this and leave it for another PR, I'm happy to do so, since it seems like this is the last unresolved thread for this PR. I don't want to hold this up unnecessarily.